File: Strategy.FSharp/Body.fs
namespace Strategy.FSharp

open Godot
open Godot.Collections

type Body() =
    inherit Node2D()
    
    let mutable nodesWithColor : Array<NodePath> = Array<NodePath>() 

    member this.NodesWithColor
        with get () = nodesWithColor
        and set value = nodesWithColor <- value
    
    override this._Ready() =
        for node in nodesWithColor do
            let node = this.GetNode(node) :?> Node2D
            node.Material <- node.Material.Duplicate() :?> Material

    member this.SetBodyRotation(rotation: float32) = this.GlobalRotationDegrees <- rotation

    member this.SetWeaponRotation(rotation: float32) =
        let weapon =
            this.GetNode(new NodePath("Weapon")) :?> Node2D

        weapon.GlobalRotationDegrees <- rotation

File: Body.cs
using Godot;
using Godot.Collections;

namespace Strategy;

[Tool]
public partial class Body : FSharp.Body
{
    [Export((PropertyHint)35, "Node2D")]
    private new Array<NodePath> NodesWithColor
    {
        get => base.NodesWithColor;
        set => base.NodesWithColor = value;
    }

    /// <inheritdoc />
    public override void _Ready()
    {
        base._Ready();
    }
}
